autoendpoint/src/routes/webpush.rs

```diff
@@ -12,15 +13,29 @@ use actix_web::HttpResponse;
 pub async fn webpush_route(
     notification: Notification,
     routers: Routers,
+    state: Data<ServerState>,
 ) -> ApiResult<HttpResponse> {
     let router = routers.get(
         RouterType::from_str(&notification.subscription.user.router_type)
             .map_err(|_| ApiErrorKind::InvalidRouterType)?,
     );
 
-    let response = router.route_notification(&notification).await?;
-
-    Ok(response.into())
+    match router.route_notification(&notification).await {
+        Ok(v) => Ok(v.into()),
+        Err(e) => {
+            if !e.kind.user_still_valid() {
+                // The user record is no longer valid. We should remove it so that the client
+                // can attempt to recover. We don't care if this succeeds or fails at this point.
+                let _ = drop_user(
```

Comment:
I thought we already had a `drop_user` code path like this already but I couldn't find it last week, but I just found it: it's the `remove_user` call in `handle_error`, only triggered on `RouterError::NotFound`. The apns router has its own custom `handle_error` with its own `remove_user` call.

So it seems this change isn't necessary.

Reply:
Yeah, I agree. I don't think we need the explicit `user_still_valid()` since we're now returning the correct error.